### PR TITLE
add check to generate migrations list

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add a step to CI pipeline to have a list of migration files. This will trigger a merge conflict if there are conflicts in the migration files.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44563

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: db/migrations/migration.list [hunk 1]
```diff
@@ -0,0 +1,352 @@
+db/migrations/mysql/000001_create_teams.down.sql
```
Comment (agnivade) [inline db/migrations/migration.list:1]: Maybe worth adding a comment here explaining the purpose of this file?

File: Makefile [hunk 2]
```diff
@@ -714,4 +714,6 @@ help:
 
 migrations-extract:
 	@echo Listing migration files
-	find db/migrations -maxdepth 2 -mindepth 2 | sort > db/migrations/migration.list
+	rm -f db/migrations/migrations.list
```
Comment (noxer) [inline Makefile:717]: Why is it deleted here when it's overwritten in the next line anyway?

**Review by noxer — CHANGES_REQUESTED** [review]
One small oddity, nothing serious. Thanks.

File: Makefile (hunk 2 above)
Comment (isacikgoz) [reply Makefile]: Ah, yeah need to remove that line. Thanks!

**Review by noxer — APPROVED** [review]
LGTM, thanks!

**Review by agnivade — APPROVED** [review]

**Pull request merged**